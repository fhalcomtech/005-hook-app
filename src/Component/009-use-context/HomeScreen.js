import React, {useContext} from 'react';
import {UserContext} from "./UserContext";

const HomeScreen = () => {
    const {user, setUser} = useContext(UserContext);
    const {name, lastname} = user;
    return (
        <div>
            <h1>Home Screen</h1>
            <p>
                Welcome {`${name} ${lastname}`}
            </p>
        </div>
    );
};

export default HomeScreen;
