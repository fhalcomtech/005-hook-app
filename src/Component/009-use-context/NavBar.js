import React from 'react';
import {Link} from "react-router-dom";

const NavBar = () => {
    return (
        <nav className="navbar bg-dark navbar-expand-lg">
            <ul className="navbar-nav">
                <li className="nav-item"><Link to={"/"} className="nav-link">Home</Link></li>
                <li className="nav-item"><Link to={"/login"} className="nav-link">Login</Link></li>
                <li className="nav-item"><Link to={"/about"} className="nav-link">About</Link></li>
            </ul>
        </nav>
    );
};

export default NavBar;
