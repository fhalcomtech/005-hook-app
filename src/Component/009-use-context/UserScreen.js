import React, {useContext} from 'react';
import {useForm} from "../../hooks/useForm";
import {UserContext} from "./UserContext";

const UserScreen = () => {
    const {user,setUser} = useContext(UserContext);
    const [formData, handlerInputChange, resetForm] = useForm({})
    const {name, lastname} = formData;
    const handlerSubmit = (e) => {
        e.preventDefault();
        setUser({
            id:(new Date()).getTime(),
            name,
            lastname
        })
        resetForm();
    }
    return (
        <div className="container">
            <form onSubmit={handlerSubmit} className="form">
                <input className="form-control mt-2" type="text" value={name} placeholder="name" onChange={handlerInputChange} name="name"/>
                <input className="form-control mt-2" type="text" value={lastname} placeholder="lastname" onChange={handlerInputChange} name="lastname"/>
                <button type="submit" className="btn btn-primary mt-2">Update</button>
            </form>
        </div>
    );
};

export default UserScreen;
