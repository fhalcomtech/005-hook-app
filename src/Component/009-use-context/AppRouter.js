import React from 'react';
import {
    BrowserRouter as Router,
    Routes,
    Route
} from 'react-router-dom';
import AboutScreen from "./AboutScreen";
import LoginScreen from "./LoginScreen";
import HomeScreen from "./HomeScreen";
import NavBar from "./NavBar";
import UserScreen from "./UserScreen";
const AppRouter = () => {
    return (
        <Router>
                <NavBar/>
                <Routes>
                    <Route path="/"     element={<HomeScreen/>}/>
                    <Route path="about" element={<AboutScreen/>}/>
                    <Route path="login" element={<LoginScreen/>}/>
                    <Route path="user"  element={<UserScreen/>}/>
                    <Route path="home"  element={<HomeScreen/>}/>
                </Routes>
        </Router>
    );
};

export default AppRouter;
