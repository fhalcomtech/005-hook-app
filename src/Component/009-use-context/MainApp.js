import React from 'react';
import AppRouter from "./AppRouter";
import {UserContext} from "./UserContext";


const MainApp = () => {return (
    <UserContext.Provider>
        <AppRouter/>
    </UserContext.Provider>
);};

export default MainApp;
