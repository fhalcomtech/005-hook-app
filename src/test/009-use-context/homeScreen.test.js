import React from "react"
import { mount } from "enzyme"
import HomeScreen from "./../../Component/009-use-context/HomeScreen"
import { UserContext } from "../../Component/009-use-context/UserContext"
describe('Test HomeScreen component', () => {
    const user = {name: "Fidel", lastname: "Lascano", edad:37}
    const setUser = jest.fn();
    const wrapper = mount(
    <UserContext.Provider value={{user, setUser:setUser}}>
        <HomeScreen/>
    </UserContext.Provider>
    )
    test('Test HomeScreen must be match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
     })
37})