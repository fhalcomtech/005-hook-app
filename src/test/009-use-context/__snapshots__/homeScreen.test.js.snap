// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HomeScreen component Test HomeScreen must be match snapshot 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <p>
    Welcome 
    Fidel Lascano
  </p>
</div>
`;
